public/script.js

```diff
@@ -1253,6 +1254,16 @@ async function getStatusTextgen() {
                 const data = await response.json();
                 if (data) {
                     const { chat_template, chat_template_hash } = data;
+                    if (wantsContextSize && "default_generation_settings" in data) {
+                        const backend_max_context = data["default_generation_settings"]["n_ctx"];
+                        if (max_context !== backend_max_context) {
+                            console.log(`Auto-switching max context from ${max_context} to ${backend_max_context}`);
+                            toastr.info(`Context Size Changed: ${max_context} ⇒ ${backend_max_context}`);
+                            max_context = backend_max_context;
```

Comment:
I tested that with the unlocked checkbox unchecked. When I connected, it *checked* the box and then set the context size to 32k rather than capping it at 8k.

Reply:
I added a `keep_context_lock` flag to `setGenerationParamsFromPreset()` which will cap the values if set and any of the values exceed the locked caps.